From 55580d6ca29d81eeec1e6ce982e52250f16ff167 Mon Sep 17 00:00:00 2001
From: eric-ch <eric.chanudet@gmail.com>
Date: Thu, 15 Jan 2015 17:09:21 +0000
Subject: [PATCH 9/9] Pass the correct format for cdrom atapi passthrough.

---
 src/qemu-device.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/qemu-device.c b/src/qemu-device.c
index 9055d4f..9fe6d69 100644
--- a/src/qemu-device.c
+++ b/src/qemu-device.c
@@ -129,8 +129,9 @@ static bool cdrom_device_parse_options (struct device_model *devmodel,
     } else {
         SPAWN_ADD_ARG (devmodel, "-drive");
         SPAWN_ADD_ARG (devmodel,
-                       "file=%s,media=cdrom,if=atapi-pt,format=pt",
-                       devicepath);
+                       "file=%s,media=cdrom,if=atapi-pt,format=%s",
+                       devicepath,
+                       dm_agent_in_stubdom() ? "pt-stubdom" : "pt-posix");
 
         free (option);
     }
-- 
2.1.0

