From: Jed <lejosnej@ainfosec.com>
Date: Mon, 16 Feb 2015 14:38:42 -0500
Subject: [PATCH] call svirt-interpose instead of qemu-dm-wrapper directly when not using a stubdom

diff --git a/src/spawn.c b/src/spawn.c
index 24b54e4..b4d3c9f 100644
--- a/src/spawn.c
+++ b/src/spawn.c
@@ -30,7 +30,8 @@
 #include "util.h"
 
 #define ARGS_BATCH 10
-#define QEMU_PATH "/usr/bin/qemu-dm-wrapper"
+#define QEMU_PATH "/usr/sbin/svirt-interpose"
+/* #define QEMU_PATH "/usr/bin/qemu-dm-wrapper" */
 #define QEMU_STUBDOM_PATH "/usr/bin/qemu-system-i386"
 #define IOEMU_PATH "/usr/sbin/svirt-interpose"
 #define IOEMU_STUBDOM_PATH "/usr/lib/xen/bin/qemu-dm"
