From: Jed <lejosnej@ainfosec.com>
Date: Wed, 4 Mar 2015 15:43:42 -0500
Subject: [PATCH] add an RPC to wakeup qemu

diff --git a/libdmbus/src/rpc_definitions.m4 b/libdmbus/src/rpc_definitions.m4
index ea7f1f0..11f2e41 100644
--- a/src/rpc_definitions.m4
+++ b/src/rpc_definitions.m4
@@ -97,6 +97,7 @@ DEFINE_MESSAGE(21, dom0_input_event, uint16_t type,
                                      int32_t value)
 DEFINE_MESSAGE(24, input_config_reset, uint8_t slot)
 DEFINE_MESSAGE(25, input_config, InputConfig c)
+DEFINE_MESSAGE(26, input_wakeup)
 
 DEFINE_IN_RPC_NO_RETURN(switcher_abs)
 DEFINE_IN_RPC_NO_RETURN(switcher_pvm_domid)
@@ -106,6 +107,7 @@ DEFINE_IN_RPC_NO_RETURN(switcher_shutdown)
 DEFINE_OUT_RPC(dom0_input_event)
 DEFINE_OUT_RPC(input_config)
 DEFINE_OUT_RPC(input_config_reset)
+DEFINE_OUT_RPC(input_wakeup)
 
 # Common message
 DEFINE_MESSAGE(23, device_model_ready)
