From: Jed <lejosnej@ainfosec.com>
Date: Mon, 16 Feb 2015 14:53:42 -0500
Subject: [PATCH] svirt-interpose: call the new qemu wrapper

diff --git a/svirt-interpose.c b/svirt-interpose.c
index c9353a7..825964e 100644
--- a/svirt-interpose.c
+++ b/svirt-interpose.c
@@ -30,7 +30,8 @@
 
 #define BUF_SIZE      256
 #define LOCAL_DOMAINS "/local/domain"
-#define QEMU          "/opt/xensource/libexec/qemu-dm-wrapper-old"
+/* #define QEMU          "/opt/xensource/libexec/qemu-dm-wrapper-old" */
+#define QEMU          "/usr/bin/qemu-dm-wrapper"
 #define RAND_DEV      "/dev/random"
 
 typedef struct data data_t;
