From d170686fe7cd2e5a7f8e1cd2b4e9d07c9ae41a0a Mon Sep 17 00:00:00 2001
From: eric-ch <eric.chanudet@gmail.com>
Date: Wed, 26 Sep 2012 19:24:06 +0100
Subject: [PATCH 04/11] [dmagent] pass to dm-agent a name for nic device

---
 xenops/dmagent.ml | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/xenops/dmagent.ml b/xenops/dmagent.ml
index 05c2970..145701c 100644
--- a/xenops/dmagent.ml
+++ b/xenops/dmagent.ml
@@ -201,7 +201,8 @@ let create_device_cdrom_pt ~trans info domid dmaid dmid kind =
 		trans.Xst.write devpath (get opt);
 		trans.Xst.write optpath kind
 
-let create_device_net ~trans domid dmaid dmid id (mac, (_, bridge), model, _) =
+let create_device_net ~trans domid dmaid dmid (mac, (_, bridge), model,
+												  is_wireless, id) =
 	let modelstr =
 		match model with
 		| None -> "rtl8139"
@@ -221,6 +222,8 @@ let create_device_net ~trans domid dmaid dmid id (mac, (_, bridge), model, _) =
 	trans.Xst.write idpath (sprintf "%u" id);
 	trans.Xst.write namepath (sprintf "%s%u" (if is_wireless then "vwif" else
 						  	  "vif") id)
+let in_extras v info =
+	List.exists (fun (e, _) -> (compare e v) == 0) info.Dm.extras
 
 (* List of all possible device *)
 let device_list =
-- 
2.1.0

